plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/button/frontend.tsx

```diff
@@ -111,11 +112,43 @@ const { state } = store< Store >(
 					'../../../../base/stores/cart-items'
 				) ) as WooStore;
 
-				// Question: should this action throw so we can capture the error here?
```

Comment:
Answer: yes I think so. Catching here to show error in the UI is a responsibility of the frontend so I moved this out of the store.